centreon-certified/bsm/bsm_connector-apiv2.lua

```diff
@@ -0,0 +1,300 @@
+    end
+  end
+
+  -- adding the HTTP POST data
+  http_request:setopt_postfields(http_post_data)
+  -- performing the HTTP request
+  http_request:perform()
+  -- collecting results
+  http_response_code = http_request:getinfo(curl.INFO_RESPONSE_CODE)
+  http_request:close()
+
+  -- Handling the return code
+  local retval = false
+  if http_response_code == 202 or http_response_code == 200 then
+    self.sc_logger:info("[EventQueue:send_data]: HTTP POST request successful: return code is " .. tostring(http_response_code))
+    retval = true
+  else
+    self.sc_logger:error("[EventQueue:send_data]: HTTP POST request FAILED, return code is " .. tostring(http_response_code) .. ". Message is: " .. tostring(http_response_body))
+  end
+  return retval
+end
+
+--------------------------------------------------------------------------------
+-- Required functions for Broker StreamConnector
+--------------------------------------------------------------------------------
+
+local queue
+
+-- Fonction init()
+function init(conf)
+  queue = EventQueue.sc_event.event.new(conf)
+end
+
+-- Fonction write()
+function write(event)
+  -- First, flush all queues if needed (too old or size too big)
+  Queue.sc_event.event.sc_flush:flush_all_queues(Queue.sc_event.event.send_data_method[1])
+
+  -- skip event if a mandatory parameter is missing
+  if Queue.sc_event.event.fail then
+    Queue.sc_event.event.sc_logger:error("Skipping event because a mandatory parameter is not set")
+    return true
+  end
+
+  -- initiate event object
+  Queue.sc_event.event.sc_event = sc_event.new(event, Queue.sc_event.event.sc_params.params, Queue.sc_event.event.sc_common, Queue.sc_event.event.sc_logger, Queue.sc_event.event.sc_broker)
+
+  -- drop event if wrong category
+  if not Queue.sc_event.event.sc_event:is_valid_category() then
+    Queue.sc_event.event.sc_logger:debug("dropping event because category is not valid. Event category is: "
+      .. tostring(Queue.sc_event.event.sc_params.params.reverse_category_mapping[Queue.sc_event.event.sc_event.event.category]))
+    return true
+  end
+
+  -- drop event if wrong element
+  if not Queue.sc_event.event.sc_event:is_valid_element() then
+    Queue.sc_event.event.sc_logger:debug("dropping event because element is not valid. Event element is: "
+      .. tostring(Queue.sc_event.event.sc_params.params.reverse_element_mapping[Queue.sc_event.event.sc_event.event.category][Queue.sc_event.event.sc_event.event.element]))
+    return true
+  end
+
+  -- drop event if it is not validated
+  if Queue.sc_event.event.sc_event:is_valid_event() then
+    queue:format_accepted_event()
+  else
+    return true
+  end
+
+  -- Since we've added an event to a specific queue, flush it if queue is full
+  Queue.sc_event.event.sc_flush:flush_queue(Queue.sc_event.event.send_data_method[1], Queue.sc_event.event.sc_event.event.category, Queue.sc_event.event.sc_event.event.element)
```

Comment:
you need to redo all this code block because it is not Queue.sc_event.event but just queue most of the time

Reply:
Fixed